bin/barclamp_lib.rb

```diff
@@ -15,66 +15,112 @@
 # limitations under the License.
 #
 
-require "rubygems"
-require "net/http"
-require "net/http/digest_auth"
-require "uri"
-require "json"
 require "getoptlong"
+require "active_support/all"
 
-@debug = false
-@hostname = ENV["CROWBAR_IP"]
-@hostname = "127.0.0.1" unless @hostname
+@hostname = ENV["CROWBAR_IP"] || "127.0.0.1"
 @port = ENV["CROWBAR_PORT"] || 80
-@headers = {
-  "Accept" => "application/json",
-  "Content-Type" => "application/json"
-}
-@data = ""
+
+@credentials = false
+@debug = false
 @allow_zero_args = false
 @timeout = 500
 @crowbar_key_file = "/etc/crowbar.install.key"
 
-#
-# Parsing options can be added by adding to this list before calling opt_parse
-#
+@data = ""
+
+@mapped_commands = [
+  "api_help",
+  "transition",
+  "show",
+  "delete",
+  "list",
+  "elements",
+  "element_node",
+  "proposal"
+]
+
 @options = [
-    [["--help", "-h", GetoptLong::NO_ARGUMENT], "--help or -h - help"],
-    [["--username", "-U", GetoptLong::REQUIRED_ARGUMENT], "--username <username> or -U <username>  - specifies the username"],
-    [["--password", "-P", GetoptLong::REQUIRED_ARGUMENT], "--password <password> or -P <password>  - specifies the password"],
-    [["--hostname", "-n", GetoptLong::REQUIRED_ARGUMENT], "--hostname <name or ip> or -n <name or ip>  - specifies the destination server"],
-    [["--port", "-p", GetoptLong::REQUIRED_ARGUMENT], "--port <port> or -p <port> - specifies the destination server port"],
-    [["--debug", "-d", GetoptLong::NO_ARGUMENT], "--debug or -d - turns on debugging information"],
-    [["--data", GetoptLong::REQUIRED_ARGUMENT], "--data <data> - used by create or edit as data (must be in json format)"],
-    [["--file", GetoptLong::REQUIRED_ARGUMENT], "--file <file> - used by create or edit as data when read from a file (must be in json format)"],
-    [["--timeout", GetoptLong::REQUIRED_ARGUMENT], "--timeout <seconds> - timeout in seconds for read http reads"]
+  [["--help", "-h", GetoptLong::NO_ARGUMENT], "--help or -h - help"],
+  [["--username", "-U", GetoptLong::REQUIRED_ARGUMENT], "--username <username> or -U <username>  - specifies the username"],
+  [["--password", "-P", GetoptLong::REQUIRED_ARGUMENT], "--password <password> or -P <password>  - specifies the password"],
+  [["--hostname", "-n", GetoptLong::REQUIRED_ARGUMENT], "--hostname <name or ip> or -n <name or ip>  - specifies the destination server"],
+  [["--port", "-p", GetoptLong::REQUIRED_ARGUMENT], "--port <port> or -p <port> - specifies the destination server port"],
+  [["--debug", "-d", GetoptLong::NO_ARGUMENT], "--debug or -d - turns on debugging information"],
+  [["--data", GetoptLong::REQUIRED_ARGUMENT], "--data <data> - used by create or edit as data (must be in json format)"],
```

Comment:
Metrics/LineLength: Line is too long. [121/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)